packages/apps/plugins/plugin-thread/src/ThreadPlugin.tsx

```diff
@@ -195,11 +208,72 @@ export const ThreadPlugin = (): PluginDefinition<ThreadPluginProvides> => {
         builder: (plugins) => {
           const client = resolvePlugin(plugins, parseClientPlugin)?.provides.client;
           const dispatch = resolvePlugin(plugins, parseIntentPlugin)?.provides.intent.dispatch;
-          if (!client || !dispatch) {
+          const metadataResolver = resolvePlugin(plugins, parseMetadataResolverPlugin)?.provides.metadata.resolver;
+
+          if (!client || !dispatch || !metadataResolver) {
             return [];
           }
 
           return [
+            createExtension({
+              id: 'comments-for-subject',
+              resolver: ({ id }) => {
+                if (!id.endsWith('~comments')) {
+                  return;
+                }
+
+                // TODO(Zan): Find util (or make one)
+                const docId = id.split('~').at(0);
+                const [spaceId, objectId] = docId?.split(':') ?? [];
+                const space = client.spaces.get().find((space) => space.id === spaceId);
+                const doc = space?.db.getObjectById(objectId);
+
+                if (!doc || !docId) {
+                  return;
+                }
+
+                const docMeta = metadataResolver(getTypename(doc) ?? '');
+                const label = docMeta.label?.(doc) || doc.name || docMeta.placeholder;
```

Comment:
see #7300 to ensure there's values for these things and consider whether you want the node to resolve or not if metadata resolver is missing for the object